### ISSUE TITLE
Make it easier to avoid backing up transient data like caches

### ISSUE THREAD

**Issue by thomaseizinger** [photoprism/photoprism#919]
Cache directories are usually pointless to include in a backup.

I am also a big fan of rolling with the default configuration of software. That has several advantages like:

- avoiding unnecessary complexity in the setup process
- potential discrepancies when the same program is started from different places (f.e. my systemd backup unit is different from the unit that starts the web server but both need the same configuration)
- easier migration to new versions
- etc

In the default configuration, photoprism's cache directory seems to default to `/var/lib/photoprism/originals/.photoprism/storage/cache`.

This is sub-optimal because it means I cannot just point my backup tool to `/var/lib/photoprism` or `/var/lib/photoprism/originals`.

What are people's thoughts on:

- Making use of the XDG directory [specification](https://wiki.archlinux.org/index.php/XDG_Base_Directory) to automatically put caches into `XDG_CACHE_HOME`
- Making use of the `CACHEDIR.TAG` [specification](https://bford.info/cachedir/) to automatically tag cache directories
- other ideas?

What I would like to highlight is the importance of not needing additional configuration for this to work:

In both of the above suggestion solutions, the enduser (myself) does not need to concern themselves with where the cache directories are and how many of them there are. Photoprism can even transparently add or split the current cache directories without the enduser needing to concern themselves with this change f.e.

**Comment by alexislefebvre** [comment]
This is interesting, thanks.

How did you install PhotoPrism? Without Docker?

Some projets added that file too, like Matomo: https://github.com/matomo-org/matomo/issues/12754

**Comment by thomaseizinger** [comment]
> How did you install PhotoPrism? Without Docker?

I am running photoprism on Manjaro Linux compiled from source so that I can manage it (and its dependencies like MariaDB) using systemd.

**Comment by lastzero** [comment]
As a quick fix, you may specify a separate storage folder so that PhotoPrism doesn't default to using a hidden folder in your originals path.

**Comment by thomaseizinger** [comment]
> As a quick fix, you may specify a separate storage folder so that PhotoPrism doesn't default to using a hidden folder in your originals path.

Yes, that is what I am thinking of doing (currently I am excluding the path manually from the backup). If I specify it in the config file, it should still be able to run photoprism from different contexts with the same configuration as long as it automatically discovers the config file :)

Are you open to any of the other solutions though? What are your thoughts on platform conventions like [XDG](https://wiki.archlinux.org/index.php/XDG_Base_Directory) or the [MacOS directories](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html#//apple_ref/doc/uid/TP40010672-CH10-SW1)?

**Comment by alexislefebvre** [comment]
Does XDG is used when running a service as `root`?

**Comment by thomaseizinger** [comment]
> Does XDG is used when running a service as `root`?

I don't know actually but I am not running photoprism as root. I created a dedicated user `photoprism` that it is running under.

**Comment by graciousgrey** [comment]
Is this still relevant as we have a separate storage directory by default now?

**Comment by thomaseizinger** [comment]
> Is this still relevant as we have a separate storage directory by default now? 

I will investigate! Currently on vacation but will come back to you within the next 2 weeks!

**Comment by graciousgrey** [comment]
No need to hurry. Enjoy your vacation :)!
